### ISSUE TITLE
With Python 3.14, pyright checks templatelib.py and complains if --pythonversion < 3.14.

### ISSUE THREAD

**Issue by Torom** [microsoft/pyright#11003]
When I use pyright under Python 3.14 with a Python file that contains any `str`, pyright checks the file `lib/python3.14/string/templatelib.py` and complains about the template strings there because `--pythonversion` is set to a Python version < 3.14.

Test code `a.py`:
```python
"hello world"
```
`pyright --pythonversion 3.13 a.py` leads to:
```
/home/torsten/.pyenv/versions/3.14.0/lib/python3.14/string/templatelib.py
  /home/torsten/.pyenv/versions/3.14.0/lib/python3.14/string/templatelib.py:3:5 - error: Template string literals (t-strings) require Python 3.14 or newer
  /home/torsten/.pyenv/versions/3.14.0/lib/python3.14/string/templatelib.py:5:24 - error: Cannot access attribute "interpolations" for class "str"
    Attribute "interpolations" is unknown (reportAttributeAccessIssue)
2 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**
cli
```
pyright --version
pyright 1.1.406
```

**Comment by erictraut** [comment]
Thanks for reporting this issue. It will be fixed in the next release.

**Comment by Bobronium** [comment]
Any viable workaround in the meanwhile?

**Comment by erictraut** [comment]
> Any viable workaround in the meanwhile?

Three potential workarounds:
1. Run pyright on Python 3.13 rather than 3.14
2. Run pyright on 3.14 but don't use pythonVersion to override the version to 3.13
3. Downgrade to pyright 1.1.401 (the version before template string support was added)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.407
